sqlglot/optimizer/simplify.py

```diff
@@ -783,6 +784,8 @@ def simplify_concat(expression):
 
     if concat_type is exp.ConcatWs:
         new_args = [sep_expr] + new_args
+    elif concat_type is exp.Concat and isinstance(expression, exp.DPipe):
```

Comment:
We can actually simplify this a bit more
```suggestion
    elif isinstance(expression, exp.DPipe):
```